### PR TITLE
Adding object model changes and Telemetry optin status from Test Window side

### PR DESCRIPTION
**Pull request by khilangoel**
Please refer TestWindow sides changes too : https://devdiv.visualstudio.com/DevDiv/VS.in%20Agile%20Testing%20IDE/_git/VS/pullrequest/88454?_a=files

### CODE REVIEW

File: test/Microsoft.TestPlatform.AcceptanceTests/TelemetryTests.cs [hunk 1]
```diff
@@ -1,157 +0,0 @@
-﻿// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (khilan2goel) [inline test/Microsoft.TestPlatform.AcceptanceTests/TelemetryTests.cs]: Removing it as of now, beacause in EnvironmentVariable logic has been removed. Will raise in next PR

**Comment by harshjain2** [comment]
Please share test plan/ scenarios verified.

**Comment by AbhitejJohn** [comment]
It appears that we are also exposing Filtering to Discovery/Run Tests with Test Cases. If we do not support Filtering for any of these please go ahead and throw a warning or error out.

**Comment by navin22** [comment]
@AbhitejJohn Clients can pass it but, is it appropriate to throw error/warning. Is it just enough to document it. As TestPlatformOptions can grow.

File: src/vstest.console/TestPlatformHelpers/TestRequestManager.cs [hunk 2]
```diff
@@ -80,7 +79,7 @@ public TestRequestManager()
                   TestRunResultAggregator.Instance,
                   TestPlatformEventSource.Instance,
                   new InferHelper(new AssemblyMetadataProvider()),
-                  MetricsPublisherFactory.GetMetricsPublisher(IsTelemetryOptedIn(), CommandLineOptions.Instance.IsDesignMode))
+                  MetricsPublisherFactory.GetMetricsPublisher(false, CommandLineOptions.Instance.IsDesignMode))
```
Comment (navin22) [inline src/vstest.console/TestPlatformHelpers/TestRequestManager.cs]: nit: Good to comment this, why we always pass false. Hence by default testplatform will not publish anything.

**Review by navin22 — APPROVED** [review]

**Review by harshjain2 — APPROVED** [review]

**Review by AbhitejJohn — CHANGES_REQUESTED** [review]

**Review by AbhitejJohn — APPROVED** [review]

**Comment by khilan2goel** [comment]
Regarding showing warning,  I have created bug Refer: https://github.com/Microsoft/vstest/issues/1189  because same is happening in CLI scenarios as well when we pass /testcasefilter along with /listtests

File: src/Microsoft.TestPlatform.VsTestConsole.TranslationLayer/DiscoveryEventsHandler.cs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel.Client;
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel.Logging;
+
+    /// <summary>
+    /// The Discovery Events Handler.
+    /// Converts the ITestDiscoveryEventsHandler to ITestDiscoveryEventsHandler2
+    /// </summary>
+    public class DiscoveryEventsHandler : ITestDiscoveryEventsHandler2
```
[the remark this answers is not in the document]
Comment (khilan2goel) [reply src/Microsoft.TestPlatform.VsTestConsole.TranslationLayer/DiscoveryEventsHandler.cs]: Done

File: test/TranslationLayer.UnitTests/VsTestConsoleRequestSenderTests.cs [hunk 4]
```diff
@@ -605,9 +605,9 @@ public void DiscoverTestsShouldCompleteWithTestMessage()
             mockHandler.Setup(mh => mh.HandleLogMessage(It.IsAny<TestMessageLevel>(), It.IsAny<string>())).Callback(
                 () => this.mockCommunicationManager.Setup(cm => cm.ReceiveMessageAsync(It.IsAny<CancellationToken>())).Returns(Task.FromResult(discoveryComplete)));
 
-            this.requestSender.DiscoverTests(new List<string>() { "1.dll" }, null, mockHandler.Object);
+            this.requestSender.DiscoverTests(new List<string>() { "1.dll" }, null, new TestPlatformOptions(), mockHandler.Object);
```
[the remark this answers is not in the document]
Comment (khilan2goel) [reply test/TranslationLayer.UnitTests/VsTestConsoleRequestSenderTests.cs]: Done

**Review by singhsarab — APPROVED** [review]
:shipit:

**Pull request merged**